tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_size.py

```diff
@@ -0,0 +1,97 @@
+"""
+EOF Container, test custom_size field for sections
+"""
+
+from enum import Enum
+
+import pytest
+
+from ethereum_test_tools import EOFTestFiller
+from ethereum_test_tools import Opcodes as Op
+from ethereum_test_tools.eof.v1 import Container, EOFException, Section, SectionKind
+
+from .. import EOF_FORK_NAME
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-3540.md"
+REFERENCE_SPEC_VERSION = "8dcb0a8c1c0102c87224308028632cc986a61183"
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+
+class SectionSize(Enum):
+    """
+    Enum for the section size
+    """
+
+    NORMAL = 1
+    UNDERSIZE = 2
+    OVERSIZE = 3
+
+
+@pytest.mark.parametrize(
+    "section_kind, section_size, exception",
+    [
+        (SectionKind.DATA, SectionSize.NORMAL, None),
+        (SectionKind.DATA, SectionSize.UNDERSIZE, EOFException.INVALID_SECTION_BODIES_SIZE),
+        (SectionKind.DATA, SectionSize.OVERSIZE, None),
+        (SectionKind.CODE, SectionSize.UNDERSIZE, EOFException.INVALID_SECTION_BODIES_SIZE),
+        (SectionKind.CODE, SectionSize.OVERSIZE, EOFException.INVALID_SECTION_BODIES_SIZE),
+        (SectionKind.CODE, SectionSize.NORMAL, None),
+        (SectionKind.TYPE, SectionSize.UNDERSIZE, EOFException.INVALID_TYPE_SECTION_SIZE),
+        (SectionKind.TYPE, SectionSize.OVERSIZE, EOFException.INVALID_SECTION_BODIES_SIZE),
+        (SectionKind.TYPE, SectionSize.NORMAL, None),
+    ],
+)
+def test_section_size(
+    eof_test: EOFTestFiller,
+    section_size: SectionSize,
+    section_kind: SectionKind,
+    exception: EOFException,
+):
+    """
+    Test custom_size is auto, more or less then the actual size of the section
+    """
+    size = (
+        0
+        if section_size == SectionSize.NORMAL
+        else 100
+        if section_size == SectionSize.OVERSIZE
+        else 2
+    )
+
+    eof_code = Container()
+
+    if section_size != SectionSize.NORMAL and section_kind == SectionKind.TYPE:
+        eof_code.sections.append(
+            Section(
+                kind=SectionKind.TYPE,
+                data="0x00800001",
+                custom_size=size,
```

Comment:
```suggestion
                custom_size=section_size,
```
After applying the first comment.

Same for the other two lines below.